src/lib/viewers/doc/DocPreloader.js

```diff
@@ -113,18 +148,37 @@ class DocPreloader extends EventEmitter {
             return;
         }
 
-        this.restoreScrollPosition();
         this.unbindDOMListeners();
+        this.restoreScrollPosition();
+        this.wrapperEl.classList.add(CLASS_IS_TRANSPARENT);
+
+        // Cleanup preload DOM after fade out
+        this.wrapperEl.addEventListener('transitionend', this.cleanupPreload);
```

Comment:
Here we fade out the preload image into the real page and then remove the preload elements. This seems to be a better experience than swapping out the preload instantaneously.